### PR TITLE
esp32-s2: wifi enhancement to include countrycode

### PR DESCRIPTION
**Pull request by BennyE**
This adds support to access the country code as announced by the Wi-Fi access point (*if* it announces it). 

The workaround via esp_wifi_get_country() is necessary as esp_wifi_sta_get_ap_info() returns wifi_ap_record_t where wifi_country_t is empty. Reference to ESP-IDF bug report/issue: https://github.com/espressif/esp-idf/issues/6267 

win10-wsl-screenshot

The discussion with @tannewt happened on this commit (so that there is more context, as questions are associated to line numbers):
https://github.com/BennyE/circuitpython/commit/ae3b4408350e5623ef769a2860984d2d8937be59 
(next time, I'll do this via a draft PR - Thank you! :))

As I haven't added new objects, it should not be necessary to add something to the gc - but I'd love to hear @dhalbert feedback. :)

### CODE REVIEW

File: ports/esp32s2/common-hal/wifi/Network.c [hunk 1]
```diff
@@ -48,3 +48,10 @@ mp_obj_t common_hal_wifi_network_get_rssi(wifi_network_obj_t *self) {
 mp_obj_t common_hal_wifi_network_get_channel(wifi_network_obj_t *self) {
     return mp_obj_new_int(self->record.primary);
 }
+
+mp_obj_t common_hal_wifi_network_get_country(wifi_network_obj_t *self) {
+    const char* cstr = (const char*) self->record.country.cc;
+        // We know that we only want the CC thus limiting to two chars
```
Comment (tannewt) [inline ports/esp32s2/common-hal/wifi/Network.c:54]: Did you get the 6 length issue sorted out?

File: ports/esp32s2/common-hal/wifi/Radio.c [hunk 2]
```diff
@@ -194,6 +198,20 @@ mp_obj_t common_hal_wifi_radio_get_ap_info(wifi_radio_obj_t *self) {
     if (esp_wifi_sta_get_ap_info(&self->ap_info.record) != ESP_OK){
         return mp_const_none;
     } else {
+        // The struct member appears to be <null> (not NULL!), I don't know how to properly test for it.
+        // When the ESP-IDF starts working fine (when their bugfix is available), this "if" wouldn't trigger.
+        // Note: It is possible that Wi-Fi APs don't have a CC set, then even after this workaround
+        //       the element would remain empty.
+        if (strlen(self->ap_info.record.country.cc) == 0) {
```
Comment (tannewt) [inline ports/esp32s2/common-hal/wifi/Radio.c:205]: This test is fine. I don't think you need the comment above. Instead you can put the comment below there and remove the log lines.

File: ports/esp32s2/common-hal/wifi/Radio.c [hunk 3]
```diff
@@ -42,6 +42,10 @@
 
 #define MAC_ADDRESS_LENGTH 6
 
+#include "components/log/include/esp_log.h"
+
+static const char* TAG = "wifi";
```
Comment (tannewt) [inline ports/esp32s2/common-hal/wifi/Radio.c:47]: No need to keep this in. Please remove.

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a few requests.

File: ports/esp32s2/common-hal/wifi/Network.c (hunk 1 above)
Comment (BennyE) [reply ports/esp32s2/common-hal/wifi/Network.c:54]: This is how it looks like if strlen(cstr) is used. I don't really know what to do about it.

>>> wifi.radio.ap_info.country
'DE \x01\r\x14'

File: ports/esp32s2/common-hal/wifi/Radio.c (hunk 2 above)
Comment (BennyE) [reply ports/esp32s2/common-hal/wifi/Radio.c:205]: Ok, will do

File: ports/esp32s2/common-hal/wifi/Network.c (hunk 1 above)
Comment (BennyE) [reply ports/esp32s2/common-hal/wifi/Network.c:54]: I understood what is happening here: This is the full country element, not just the .cc portion! 
D (0x44) E (0x45) " " (0x20 - interpreted as "whitespace" although this stands for "Any Environment" in Wi-Fi context)
image

**Comment by BennyE** [comment]
All suggested changes implemented. I also fixed the trailing line/whitespace that caused the CI to be unhappy.

grafik

File: ports/esp32s2/common-hal/wifi/Network.c (hunk 1 above)
Comment (BennyE) [reply ports/esp32s2/common-hal/wifi/Network.c]: The space character is actually correct to refer to the "Any Environment" (0x20) opposed to just "Indoor" or "Outdoor". As per feedback by espressif and also after reviewing different documents, this appears to be correct. As strlen() reads until a \0 character, I suggest we stick to the 2 to read just the country code. I've updated the comment accordingly and no longer refer to a bug. It is mainly my misinterpretation/lack of knowledge here.

**Review by tannewt — APPROVED** [review]
Looks good to me! Thank you!

**Pull request merged**